lib/src/combined_wrappers/combined_iterable.dart

```diff
@@ -45,17 +45,30 @@ class _CombinedIterator<T> implements Iterator<T> {
   /// avoid instantiating unnecessary iterators.
   final Iterator<Iterator<T>> _iterators;
 
-  _CombinedIterator(this._iterators);
+  /// The current iterator in [_iterators], or `null` if done iterating.
```

Comment:
I don't quite follow why we need this. Why did we have to `_advance()` in the constructor?

Reply:
You are always supposed to call `moveNext()` before asking for `current`. The behavior is undefined and we previously relied on the fact that it would return `null`. This is no longer possible to rely on with NNBD, and its better not to anyways :). Now we properly `moveNext` instead.